File: frontend/demo/component/avatar/avatar-menu-bar.ts
import 'Frontend/demo/init'; // hidden-source-line

import { html, LitElement } from 'lit';
import { customElement, state } from 'lit/decorators.js';
import '@vaadin/avatar';
import '@vaadin/menu-bar';
import type { MenuBarItem } from '@vaadin/menu-bar';
import { applyTheme } from 'Frontend/generated/theme';
import { getPeople } from 'Frontend/demo/domain/DataService';
import type Person from 'Frontend/generated/com/vaadin/demo/domain/Person';

@customElement('avatar-menu-bar')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  @state()
  private menuBarItems: MenuBarItem[] = [];

  @state()
  private person: Person | undefined;

  // tag::snippet[]
  protected override async firstUpdated() {
    const { people } = await getPeople({ count: 1 });
    this.person = people[0];

    const avatarElement = document.createElement('vaadin-avatar');
    avatarElement.name = `${this.person?.firstName} ${this.person?.lastName}`;
    avatarElement.img = this.person?.pictureUrl;

    this.menuBarItems = [
      {
        component: avatarElement,
        children: [
          {
            text: 'Profile',
          },
          {
            text: 'Settings',
          },
          {
            text: 'Help',
          },
          {
            text: 'Sign out',
          },
        ],
      },
    ];
  }

  protected override render() {
    return html`
      <vaadin-menu-bar .items="${this.menuBarItems}" theme="tertiary-inline"></vaadin-menu-bar>
    `;
  }
  // end::snippet[]
}

File: src/main/java/com/vaadin/demo/component/avatar/AvatarMenuBar.java
package com.vaadin.demo.component.avatar;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.demo.domain.DataService;
import com.vaadin.demo.domain.Person;
import com.vaadin.flow.component.avatar.Avatar;
import com.vaadin.flow.component.contextmenu.MenuItem;
import com.vaadin.flow.component.contextmenu.SubMenu;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.menubar.MenuBar;
import com.vaadin.flow.component.menubar.MenuBarVariant;
import com.vaadin.flow.router.Route;

@Route("avatar-menu-bar")
public class AvatarMenuBar extends Div {

  private Person person = DataService.getPeople(1).get(0);

  public AvatarMenuBar() {
    String name = person.getFirstName() + " " + person.getLastName();
    String pictureUrl = person.getPictureUrl();

    // tag::snippet[]
    Avatar avatar = new Avatar(name);
    avatar.setImage(pictureUrl);

    MenuBar menuBar = new MenuBar();
    menuBar.addThemeVariants(MenuBarVariant.LUMO_TERTIARY_INLINE);

    MenuItem menuItem = menuBar.addItem(avatar);
    SubMenu subMenu = menuItem.getSubMenu();
    subMenu.addItem("Profile");
    subMenu.addItem("Settings");
    subMenu.addItem("Help");
    subMenu.addItem("Sign out");
    // end::snippet[]

    add(menuBar);
  }

  public static class Exporter extends DemoExporter<AvatarMenuBar> { // hidden-source-line
  } // hidden-source-line
}
